lib/dockerregistry/inventory.go

```diff
@@ -119,9 +120,62 @@ func ParseManifestFromFile(filePath string) Manifest {
 func ParseManifest(bytes []byte) (Manifest, error) {
 	var m Manifest
 	err := yaml.UnmarshalStrict(bytes, &m)
+	if err != nil {
+		return m, err
+	}
+
+	err = m.Validate()
 	return m, err
 }
 
+// Validate checks for semantic errors in the yaml fields (the structure of the
+// yaml is checked during unmarshaling).
+func (m Manifest) Validate() error {
+	err := validateImages(m.Images)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateImages(images []Image) error {
+	var err error
```

Comment:
If you'll refactor the error handling I commented below, you could get rid off this variable declaration.